graphics/wincursor.cpp

```diff
@@ -151,8 +150,10 @@ bool WinCursor::readFromStream(Common::SeekableReadStream &stream) {
 	if (numColors == 0)
 		numColors = 1 << bitsPerPixel;
 
+	// Skip number of important colors
+	stream.skip(4);
+
 	// Reading the palette
-	stream.seek(40 + 4);
```

Comment:
I hope, this does not break existing things :)

Reply:
Some explanation of how this works: In EXE resources, the format of a cursor is 4 bytes for the hotspot X and Y followed by the DIB (which starts with a `BITMAPINFOHEADER`).  In a CUR file, or a subresource of an ANI file which is in CUR format, there is no hotspot prefix, it simply starts with `BITMAPINFOHEADER`, and the hotspot location is stored in the `ICONDIR` structure in the directory instead.  The first field of `BITMAPINFOHEADER` is a 32-bit size, which is why the 40 is supposed to be there.  In theory, it can be other values for extended bitmap headers, but we don't support those in this code.

So, I put moved the DIB loading into its own function here, which is called with the stream positioned at the start of the DIB, and moved the hotspot loading from EXE cursor resources into `WinCursorGroup::createCursorGroup`.  I regression tested that with Reah and it works fine.

The problem with the seek though is that the 40 + 4 is supposed to be seeking to the end of the `BITMAPINFOHEADER`, but that only works if the `BITMAPINFOHEADER` is located at position 4, but in the CUR file substream it is at position 0.  So, this uses a relative-position seek instead to seek to the correct location regardless of where it started reading.